kmcd-prelude.md

```diff
@@ -144,6 +143,134 @@ module KMCD-PRELUDE
          transact ADMIN End . initOut "gold" "Bobby"
          .MCDSteps
       [macro]
+endmodule
+```
 
+```k
+module KMCD-GEN
+    imports KMCD-PRELUDE
+
+    configuration
+      <kmcd-random>
+        <kmcd-properties/>
+        <kmcd-gen>
+          <random> $RANDOMSEED:Int </random>
+          <generator-depth-bound> $GENDEPTH:DepthBound </generator-depth-bound>
+          <generator-next> 0 </generator-next>
+          <generator-current> 0 </generator-current>
+          <generator-remainder> .GenStep </generator-remainder>
+          <generators>
+            <generator multiplicity="*" type="Map">
+              <generator-id> 0 </generator-id>
+              <generator-steps> .GenStep </generator-steps>
+            </generator>
+          </generators>
+          <end-gen> .GenStep </end-gen>
+        </kmcd-gen>
+      </kmcd-random>
+
+    syntax DepthBound ::= Int | "*"
+                        | decrement ( DepthBound ) [function]
+ // ---------------------------------------------------------
+    rule decrement(*) => *
+    rule decrement(N) => N -Int 1
```

Comment:
We might want to add the condition that `N >Int 0`, to disallow improper usage. And perhaps also another rule `rule decrement(N) => 0 if N <=Int 0` to avoid partiality in this case.

Reply:
Will do.